functions/go/_template/README.md

```diff
@@ -6,23 +6,39 @@
 
 Explain what this function does in one or two sentences.
```

Comment:
The one-sentence overviews are critical to get right. Currently they are missing the mark.

https://catalog.kpt.dev/

Examples:

- "Ensure a name substring is part of the name." This is too vague. What you want to say probably: Ensures that a resource name contains the given substring.
- "Fix resources and make them compatible with kpt 1.0." This is tautological. What does "fix" mean?
- "Format resources in a directory." This contains reference to "directory" which is both extraneous (all functions act on some resource in a directory) and misleading (hydration is recursive). It's also tautological: "format function formats"
- "Upsert a resource to the input list of resources." What's "input list of resources"? It's also extraneous since all functions operate on input resources.

Please generalize these recommendations here, and follow up on addressing them for all functions.

Reply:
@phanimarupaka

Reply:
The 1-line description in catalog landing page (https://catalog.kpt.dev/) is generated from `metadata.yaml` description.

Reply:
The changes should be in separate PR(s). Created https://github.com/GoogleContainerTools/kpt/issues/2343 to track

Reply:
@frankfarzan PTAL https://github.com/GoogleContainerTools/kpt-functions-catalog/pull/440